engine/common/rpc/convert/execution_results_test.go

```diff
@@ -26,7 +25,25 @@ func TestConvertExecutionResult(t *testing.T) {
 	assert.Equal(t, er, converted)
 }
 
```

Comment:
Lets test the boundary conditions (especially, the upper limit of the possible value range):

```suggestion

// Tests that Protobuf conversion is reversible for old chunk format (Protocol Version <2)
// TODO(mainnet27, #6773): remove this function
func TestServiceEventCount_conversion(t *testing.T) {
	for i := uint16(0); i < 20; i++ {
		j := convert.ServiceEventCountFieldToMessage(&i)
		r := convert.MessageToServiceEventCountField(j)
		require.NotNil(t, r, "encoding value %d resulted in nil", i)
		require.Equal(t, i, *r)
	}

	for i := uint16(math.MaxUint16); i > math.MaxUint16-20; i-- {
		j := convert.ServiceEventCountFieldToMessage(&i)
		r := convert.MessageToServiceEventCountField(j)
		require.NotNil(t, r, "encoding value %d resulted in nil", i)
		require.Equal(t, i, *r)
	}
}

```
you'll notice the test failing, indicating a typo (see [my next comment](url))